modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java

```diff
@@ -516,25 +515,28 @@ static CompletableFuture<List<Set<Assignment>>> tableAssignments(
     }
 
     /**
-     * Returns table assignments for all table partitions from vault.
+     * Returns table assignments for all table partitions from meta storage locally.
      *
-     * @param vaultManager Vault manager.
+     * @param metaStorageManager Meta storage manager.
      * @param tableId Table id.
      * @param numberOfPartitions Number of partitions.
+     * @param revision Revision.
      * @return Future with table assignments as a value.
      */
     public static List<Set<Assignment>> tableAssignments(
-            VaultManager vaultManager,
+            MetaStorageManager metaStorageManager,
             int tableId,
-            int numberOfPartitions
+            int numberOfPartitions,
+            long revision
     ) {
         return IntStream.range(0, numberOfPartitions)
-                .mapToObj(i ->
-                        (Set<Assignment>) ByteUtils.fromBytes(
-                                vaultManager.get(
-                                        stablePartAssignmentsKey(new TablePartitionId(tableId, i))
-                                ).join().value())
-                )
+                .mapToObj(p -> {
+                    Entry e = metaStorageManager.getLocally(stablePartAssignmentsKey(new TablePartitionId(tableId, p)), revision);
+
+                    assert !e.empty() && !e.tombstone() : e;
```

Comment:
Because we first check the presence of assignments for one partition, and then presume that others are also present. If some partitions missing, this is some kind of corruption and we throw assertion.

Reply:
I didn't get the point. Why we treat assignments presence differently in `tableAssignmentsGetLocally` and `tableAssignments`?